src/go/k8s/pkg/resources/resource.go

```diff
@@ -1,16 +1,23 @@
-package resources
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
 
-import (
-	"context"
+// Package resources contains reconciliation logic for redpanda.vectorized.io CRD
+package resources
 
-	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-)
+import "context"
 
+// Resource decompose the reconciliation loop to specific kubernetes obejcts
 type Resource interface {
-	Obj() (client.Object, error)
-	Key() types.NamespacedName
+	// Kind returns the canonical name of the kubernetes managed resource
 	Kind() string
 
+	// Ensure gives the main cluster controller ability to reconcile parts of the
```

Comment:
Given this is an interface, I found this somehow specific. I would just write "Reconciles the resource".

Reply:
If we would like to be not that specific we should remove cluster controller too. My proposal:
> Ensure reconcile only one resource available in Kubernetes API server